environmentManager/__tests__/environmentManager.test.js

```diff
@@ -0,0 +1,281 @@
+const request = require('supertest');
+const app = require('../../app');
+
+// Mocked client to return authorization key when getTreatment is called
+jest.mock('../../sdk', () => ({
+  getSplitFactory: jest.fn((settings) => {
+    let sdk = jest.requireActual('../../sdk');
+    const factory = sdk.getSplitFactory(settings);
```

Comment:
If we create a factory without destroying it in the test teardown, Jest will not exit properly, due to open handlers in JS event-loop, and so we might not be able to detect some leaks on the SDK.

We should update this test in order to avoid jest warnings such as `A worker process has failed to exit gracefully and has been force exited.` or `Jest did not exit one second after the test run has completed.`